lib/block_view.go

```diff
@@ -2737,27 +2756,49 @@ func (bav *UtxoView) _disconnectAcceptNFTBid(
 
 	// (3) Revert payments made from accepting the NFT bids.
 	if operationData.NFTPaymentUtxoKeys == nil || len(operationData.NFTPaymentUtxoKeys) == 0 {
-		return fmt.Errorf("_disconnectAcceptNFTBid: NFTPaymentUtxoKeys was nil; " +
-			"this should never happen")
+		return fmt.Errorf("_helpDisconnectNFTSold: NFTPaymentUtxoKeys was nil; this should never happen")
 	}
 	// Note: these UTXOs need to be unadded in reverse order.
 	for ii := len(operationData.NFTPaymentUtxoKeys) - 1; ii >= 0; ii-- {
 		paymentUtxoKey := operationData.NFTPaymentUtxoKeys[ii]
 		if err := bav._unAddUtxo(paymentUtxoKey); err != nil {
-			return errors.Wrapf(err, "_disconnectAcceptNFTBid: Problem unAdding utxo %v: ", paymentUtxoKey)
+			return errors.Wrapf(err, "_helpDisconnectNFTSold: Problem unAdding utxo %v: ", paymentUtxoKey)
 		}
 	}
 
-	// (4) Revert spent bidder UTXOs.
-	if operationData.NFTSpentUtxoEntries == nil || len(operationData.NFTSpentUtxoEntries) == 0 {
-		return fmt.Errorf("_disconnectAcceptNFTBid: NFTSpentUtxoEntries was nil; " +
-			"this should never happen")
-	}
-	// Note: these UTXOs need to be unspent in reverse order.
-	for ii := len(operationData.NFTSpentUtxoEntries) - 1; ii >= 0; ii-- {
-		spentUtxoEntry := operationData.NFTSpentUtxoEntries[ii]
-		if err := bav._unSpendUtxo(spentUtxoEntry); err != nil {
-			return errors.Wrapf(err, "_disconnectAcceptNFTBid: Problem unSpending utxo %v: ", spentUtxoEntry)
+	// We do not need to revert bidder UTXOs if this is an NFT Bid on a Buy Now NFT, because the bidder inputs are specified
+	// as transaction inputs as opposed to bidder inputs that are specified in the transaction metadata since the transactor
+	// and the bidder are the same in this scenario.
+	if operationData.Type != OperationTypeNFTBid {
```

Comment:
Nit: I think it's safer to write this as:
```
if operationData.Type == OperationTypeAcceptNFTBid {
...
} else if operationData.Type == OperationTypeNFTBic {
...
} else {
// error because we got a weird type
}
```